src/explore-education-statistics-frontend/src/modules/data-catalogue/DataCataloguePage.tsx

```diff
@@ -0,0 +1,208 @@
+  {
+    id: 'rel-1',
+    latestRelease: true,
+    published: '2021-06-30T11:21:17.7585345',
+    slug: 'rel-1-slug',
+    title: 'Release 1',
+  } as Release,
+  {
+    id: 'rel-3',
+    published: '2021-01-01T11:21:17.7585345',
+    slug: 'rel-3-slug',
+    title: 'Another Release',
+  } as Release,
+  {
+    id: 'rel-2',
+    published: '2021-05-30T11:21:17.7585345',
+    slug: 'rel-2-slug',
+    title: 'Release 2',
+  } as Release,
+];
+const fakeSubjectsWithDownloadFiles: SubjectWithDownloadFiles[] = [
+  {
+    id: 'subject-1',
+    name: 'Subject 1',
+    content: 'Some content here',
+    geographicLevels: ['SoYo'],
+    timePeriods: {
+      from: '2016',
+      to: '2019',
+    },
+    downloadFile: {
+      id: 'file-1',
+      extension: 'csv',
+      fileName: 'file-1.csv',
+      name: 'File 1',
+      size: '100mb',
+      type: 'Data',
+    },
+  },
+  {
+    id: 'subject-2',
+    name: 'Another Subject',
+    content: 'Some content here',
+    geographicLevels: ['SoYo'],
+    timePeriods: {
+      from: '2016',
+      to: '2019',
+    },
+    downloadFile: {
+      id: 'file-2',
+      extension: 'csv',
+      fileName: 'file-2.csv',
+      name: 'File 2',
+      size: '100mb',
+      type: 'Data',
+    },
+  },
+  {
+    id: 'subject-3',
+    name: 'Subject 3',
+    content: 'Some content here',
+    geographicLevels: ['SoYo'],
+    timePeriods: {
+      from: '2016',
+      to: '2019',
+    },
+    downloadFile: {
+      id: 'file-3',
+      extension: 'csv',
+      fileName: 'file-3.csv',
+      name: 'File 3',
+      size: '100mb',
+      type: 'Data',
+    },
+  },
+];
+
+interface Props {
+  themes: Theme[];
+}
+
+interface DataCatalogueState {
+  initialStep: number;
+  releases: Release[];
+  query: {
+    isLatestRelease: boolean;
+    releaseId: string;
+    subjects: SubjectWithDownloadFiles[];
+  };
+}
+
+const DataCataloguePage: NextPage<Props> = ({ themes }: Props) => {
+  const router = useRouter();
+
+  const [state, updateState] = useImmer<DataCatalogueState>({
+    initialStep: 1,
+    releases: [],
+    query: {
+      isLatestRelease: false,
+      releaseId: '',
+      subjects: [],
+    },
+  });
+
+  const handlePublicationStepBack = () => {
+    router.push('/data-catalogue', undefined, { shallow: true });
+  };
+
+  const handlePublicationFormSubmit: PublicationFormSubmitHandler = async () => {
+    updateState(draft => {
+      draft.releases = fakeReleases;
+    });
+  };
+
+  const handleReleaseFormSubmit: ReleaseFormSubmitHandler = async ({
+    releaseId: selectedReleaseId,
+  }) => {
+    updateState(draft => {
+      draft.query.isLatestRelease =
+        state.releases.find(release => release.id === selectedReleaseId)
+          ?.latestRelease ?? false;
+      draft.query.releaseId = selectedReleaseId;
+      draft.query.subjects = fakeSubjectsWithDownloadFiles;
+    });
+  };
+
+  const handleDownloadFormSubmit: DownloadFormSubmitHandler = ({ files }) => {
+    // DO STUFF!
+  };
+
```

Comment:
Could process.env.NODE_ENV be used?

Reply:
I've used process.env.NODE_ENV to show the the Not found page and not fetch the themes on production. It doesn't set the response code to 404 though as I'm not sure the best way to do that (pending Next 10 update), but I'm also not sure that it's necessary as it's very unlikely to be indexed as there will be no links to it until it's finished (unless there's an autogenerated site map somewhere?)

Reply:
Don't think `NODE_ENV` should be used, as I think it only has like 3 settings in Next.js - development (local), test (unit tests) and production (for builds).

`NODE_ENV` will be set to production when the frontend build is ran, meaning we won't be able to see this in any environment outside of local. I figure it would be useful for initial testing/feedback from Laura to display this in dev/test/pre-prod.

I'd say we should go with setting up the `APP_ENV` variable.

For setting the 404 status code - in `getServerSideProps` we have access to a `context.response` object that we can set the `statusCode` for.